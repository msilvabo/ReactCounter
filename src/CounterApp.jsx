import {useState} from 'react';
import PropTypes from 'prop-types';

export const CounterApp = ({value}) => {
    
    const [counter,setCounter ] = useState(value);
    const [otro,setOtro ] = useState(value + 1);
    
    const handleAdd =()=> setCounter( (c)=> c + 1);

    const handleSubs =()=> setCounter( (c)=> c - 1);

    const handleReset =()=> {
        setCounter( value);
        setOtro(value);
    }

    return(
        <>
            <h1>CounterApp</h1>
            <h2> { otro } </h2>
            <h2 data-testid='contador'> { counter } </h2>
            <button onClick={handleAdd}
            >Increment +1</button>
            <button onClick={handleSubs}
            >Decrement -1</button>
            <button aria-label='btn-reset' onClick={handleReset}
            >Reset</button>
        </>
    );
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired
}

