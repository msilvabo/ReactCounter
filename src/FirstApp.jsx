import PropTypes from 'prop-types';

const nombre = {
    message: "hola mundo",
    city: "La Paz"
};

export const FirstApp = ( {
        title = "Titulo por defecto", 
        subTitle = "Subtitulo",
        name = "Carlos"
    }) => {
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <h3>{name}</h3>
        </>
    )
}

FirstApp.propTypes={
    title: PropTypes.string.isRequired,
    valor: PropTypes.number,
    otro: PropTypes.number
}

// // FirstApp.defaultProps = {
// //     title:'Titulo por defecto',
// //     otro: 111,
// //     name: 'Carlos'
// }

/**
 * 
 * @returns String
 */
const sayHello = (country) => {
    return `Say Hello ${country} `;
}