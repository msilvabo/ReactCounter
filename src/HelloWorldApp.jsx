// const App = () => {
//     return <h1>Hola Mundo!!!</h1>
// }

// export {
//     App
// }

import React from 'react'

const HelloWorldApp = () => {
  return (
    <h1>
      Hello World
    </h1>
  )
}

export default HelloWorldApp
