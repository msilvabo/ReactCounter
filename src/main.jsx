import React from "react";
import ReactDOM from "react-dom/client";
// import { FirstApp } from "./FirstApp";
// import HelloWorldApp from "./HelloWorldApp";
import { CounterApp } from "./CounterApp";
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <HelloWorldApp /> */}
        {/* <FirstApp valor={10} otro={100} /> */}
        <CounterApp value= {100} />
    </React.StrictMode>
);