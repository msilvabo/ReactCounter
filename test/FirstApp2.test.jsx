import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('prueba en FirstApp', () => {

  const title = 'Hola, soy Goku';
  const subTitulo = 'Este es el subtitulo';

  test('deber de hacer match con el snapshot', () => {
    const {container} = render(<FirstApp title={title}/>);
    expect(container).toMatchSnapshot();
  });

  test('debe de mostrar el mensaje ""Hola, soy Goku"', () => {
    render(<FirstApp title={title}/>);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test('debe de mostrar el titulo en H1', () => {
    render(<FirstApp title={title}/>);
    expect(screen.getByRole('heading', {level: 1}).innerHTML).toBe(title);
  });
  test('debe de mostrar el subtitulo', () => {
    render(<FirstApp title={title} subTitle={subTitulo}/>);
    expect(screen.getAllByText(subTitulo).length).toBe(2);
  });
  
})
