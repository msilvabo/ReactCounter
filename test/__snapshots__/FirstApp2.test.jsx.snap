// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en FirstApp deber de hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Goku
  </h1>
  <h2>
    Subtitulo
  </h2>
  <h2>
    Subtitulo
  </h2>
  <h3>
    Carlos
  </h3>
</div>
`;
